### PR TITLE
Add os as a matrix build a variable

### PR DESCRIPTION
**Pull request by yitingw3**
Testing done: https://github.com/yitingw3/ci-configuration-examples/actions/runs/4547050649

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,10 +9,13 @@ on:
   workflow_dispatch:
 
 jobs:
-  # This workflow contains a single job called "build-v1"
   build-v1:
+    strategy:
+      matrix:
+        # Run this workflow on different os
+        os: [windows-latest, ubuntu-22.04, macos-latest]
```
Comment (mcafaro) [inline .github/workflows/ci.yml:16]: We pin to ubuntu-22.04. Should we also pin to particular versions of Windows and macOS?

**Review by mcafaro — COMMENTED** [review]
I believe we have "unlimited" minutes as a public repo but we may get caught by abuse protections running builds on Linux, Windows, and macOS 24x7 without any code changes. We'll have to see... but Windows and particularly macOS are quite a bit more expensive to run than Linux.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (yitingw3) [reply .github/workflows/ci.yml:16]: Sure, do we have a recommended version? if not, what about windows-2022 and macos-12?

**Comment by davidbuzinski** [comment]
> I believe we have "unlimited" minutes as a public repo but we may get caught by abuse protections running builds on Linux, Windows, and macOS 24x7 without any code changes. We'll have to see... but Windows and particularly macOS are quite a bit more expensive to run than Linux.

That's a good point. Looking at their [limits](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration), It doesn't seem like we're breaking any rules, and we have a good use-case, but I don't know much about how their abuse detection works.

**Review by mcafaro — APPROVED** [review]
:shipit:

**Pull request merged**